tripal/src/Access/TripalEntityAccessControlHandler.php

```diff
@@ -2,37 +2,126 @@
 
 namespace Drupal\tripal\Access;
 
-use Drupal\Core\Entity\EntityAccessControlHandler;
+use Drupal\Core\Access\AccessResultInterface;
+use Drupal\Core\Cache\CacheableMetadata;
 use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityAccessControlHandler;
 use Drupal\Core\Session\AccountInterface;
 use Drupal\Core\Access\AccessResult;
+use Drupal\tripal\Entity\TripalEntityType;
+use Drupal\user\Entity\Role;
 
 /**
- * Access controller for the Tripal Content entity.
+ * Access controller for the Tripal Content entity (i.e. TripalEntity).
  *
  * @see \Drupal\tripal\Entity\TripalEntity.
  */
 class TripalEntityAccessControlHandler extends EntityAccessControlHandler {
 
   /**
    * {@inheritdoc}
+   *
+   * Note: we do not want to call the parent here as we are doing our own thing
+   * and do not want to introduce any unexpected behaviour.
+   *
+   * If an extension module wants to override what we are doing here then they
+   * should implement one of the following hooks which take priority:
+   * hook_entity_access() or hook_ENTITY_TYPE_access().
    */
   protected function checkAccess(EntityInterface $entity, $operation, AccountInterface $account) {
+    $cacheability = new CacheableMetadata();
 
-    // Ensure that the Tripal Content Admin permission bypasses the following permissions.
+    // Ensure that the Tripal Content Admin permission bypasses
+    // the following permissions.
     if ($account->hasPermission('administer tripal content')) {
       return AccessResult::allowed();
     }
 
+    // For view, update, delete we want to use bundle-specific permissions
+    // so lets get the bundle now.
+    $entity_bundle = $entity->getType();
+
+    // Now, lets check the permission based on the bundle and operation.
```

Comment:
```suggestion
    // so let's get the bundle now.
    $entity_bundle = $entity->getType();

    // Now, let's check the permission based on the bundle and operation.
```
Minor fussy change 🙄